docs/resources/trigger.md

```diff
@@ -110,40 +110,40 @@ resource "wallarm_trigger" "attack_trigger" {
-* `template_id` - (Required) Trigger condition. A condition is a system event to be notified about. Can be:
+* `client_id` - (optional) ID of the client to apply the trigger to. The value is required for [multi-tenant scenarios][2].
+* `template_id` - (**required**) trigger condition. A condition is a system event to be notified about. Can be:
   - `user_created` for a user added to the company account in Wallarm Console
   - `attacks_exceeded` for detected attacks number exceeded the specified value
   - `hits_exceeded` for detected hits number exceeded the specified value
   - `incidents_exceeded` for detected incidents number exceeded the specified value
   - `vector_attack` for detected attack vectors number exceeded the specified value
   - `bruteforce_started` for detected attack to be identified as bruteforce
-* `enabled` - (Optional) Indicator of the trigger status. Can be: `true` for enabled trigger and `false` for disabled trigger (notifications are not sent).
-* `name` - (Optional) Trigger name.
-* `comment` - (Optional) Trigger description.
-* `filters` - (Optional) Filters for trigger conditions. Possible attributes are described below.
-* `threshold` - (Optional) Limitations for trigger conditions. Possible attributes are described below.
-* `actions` - (Optional) Trigger actions. Possible attributes are described below.
+* `enabled` - (optional) indicator of the trigger status. Can be: `true` for enabled trigger and `false` for disabled trigger (notifications are not sent).
+* `name` - (optional) Trigger name.
+* `comment` - (optional) Trigger description.
+* `filters` - (optional) Filters for trigger conditions. Possible attributes are described below.
+* `threshold` - (optional) Limitations for trigger conditions. Possible attributes are described below.
+* `actions` - (optional) Trigger actions. Possible attributes are described below.
 
 ## Filters
 
 `filters` are filters for trigger conditions. Can be:
 
-* `filter_id` - (Optional) Filter name. Can be:
+* `filter_id` - (optional) Filter name. Can be:
   - `ip_address` - IP address from which the request is sent
   - `pool` - ID of the [application](https://docs.wallarm.com/user-guides/settings/applications/) that receives the request or in which an incident is detected.
-  - `attack_type` - Type of the attack detected in the request or a type of vulnerability the request is directed to.
+  - `attack_type` - type of the attack detected in the request or a type of vulnerability the request is directed to.
   - `domain` - Domain that receives the request or in which an incident is detected.
   - `target` - Application architecture part that the attack is directed at or in which the incident is detected. Can be:
     * `Server`
     * `Client`
     * `Database`
 * `response_status` - Integer response code returned to the request.
 * `hint_tag` - Arbitrary tag of any request tuned in by a rule.
```

Comment:
?

Reply:
what is the corresponding parameter in UI?

Reply:
We don't show it in the UI, but this is what connects a brute/dirbust/BOLA trigger to the corresponding counter rule. It's transparent for users of our UI. We automatically create a counter rule using JS just before creating a trigger, and we provide the rule's random tag to the trigger.

We shouldn't copy that behavior here, however. According to Terraform's Provider Design Principles, [Resources should represent a single API object](https://developer.hashicorp.com/terraform/plugin/hashicorp-provider-design-principles#resources-should-represent-a-single-api-object). This means, we shouldn't create other resources, like counter rules, when a user creates some resource, like a brute trigger. Instead, the user should create both of them themselves.

```hcl
# here's a counter rule
resource "wallarm_rule_bruteforce_counter" "my_counter_rule" {
    action {
		type = "iequal"
		value = "example.com"
		point = {
			header = "HOST"
		}
	}
}

# here's a trigger
resource "wallarm_trigger" "my_brute_trigger" {
	template_id = "bruteforce_started"

	filters {
		filter_id = "hint_tag"
		operator = "eq"
                # and here we connect the two
		value = [wallarm_rule_bruteforce_counter.my_counter_rule.counter]
	}
	
	filters {
		filter_id = "ip_address"
		operator = "eq"
		value = ["1.1.1.1"]
	}
	
	actions {
		action_id = "mark_as_brute"
	}

	threshold = {
		period = 30
		operator = "gt"
		count = 30
	}
}
```